lib/MusicBrainz/Server/Form/Application.pm

```diff
@@ -40,7 +41,21 @@ sub validate
 
     if ($self->field('oauth_type')->value eq 'web') {
         if (!$self->field('oauth_redirect_uri')->value) {
-            $self->field('oauth_redirect_uri')->add_error(l('Redirect URL must be entered for web applications.'));
+            $self->field('oauth_redirect_uri')->add_error(
+                l('Redirect URL must be entered for web applications.')
+            );
+        } elsif ($self->field('oauth_redirect_uri')->value !~ $OAUTH_WEB_APP_REDIRECT_URI_RE) {
+            $self->field('oauth_redirect_uri')->add_error(
+                l('Redirect URL scheme must be either <code>http</code> or ' .
+                  '<code>https</code> for web applications.')
+            );
+        }
+    } elsif ($self->field('oauth_redirect_uri')->value) {
+        if ($self->field('oauth_redirect_uri')->value !~ $OAUTH_INSTALLED_APP_REDIRECT_URI_RE) {
+            $self->field('oauth_redirect_uri')->add_error(
+                l('Redirect URL scheme must be a reverse-DNS string, e.g. ' .
+                  '<code>org.example.app</code>, for installed applications.')
```

Comment:
Can we specify `://` there too? Apparently Android requires that at least and this message confused me at first.
```suggestion
                  '<code>org.example.app://</code>, for installed applications.')
```

Reply:
As `://` is technically not part of the scheme, I made a [full example](https://github.com/metabrainz/musicbrainz-server/pull/916/commits/f6d353a1a8b7384be4af93b6c9f0c02b6ffa8e33#diff-4cf7869253939843a3ad96313ecbce3dR56) instead.